external-crates/move/crates/move-analyzer/src/symbols/compilation.rs

```diff
@@ -575,65 +575,45 @@ fn merge_user_programs(
     (parsed_program_cached, typed_modules_cached)
 }
 
-/// Checks if a parsed module has been modified by comparing
-/// file hash in the module with the file hashes provided
-/// as an argument to see if module hash is included in the
-/// hashes provided. We only consider file hashes from modified
-/// files.
+/// Checks if a parsed module is modified by getting
+/// the module's file path and checking if it's included
+/// in the set of modified file paths.
 fn is_parsed_mod_modified(
     mdef: &P::ModuleDefinition,
     modified_files: &BTreeSet<PathBuf>,
-    file_hashes: Arc<BTreeMap<PathBuf, FileHash>>,
+    file_paths: Arc<BTreeMap<FileHash, PathBuf>>,
 ) -> bool {
-    !hash_included_in_file_hashes(mdef.loc.file_hash(), modified_files, file_hashes)
+    // unwrap is safe as file paths comes from mapped files
+    // that the compiler relies on when comppiling all modules
+    let mod_file_path = file_paths.get(&mdef.loc.file_hash()).unwrap();
+    modified_files.contains(mod_file_path)
 }
 
-/// Checks if a typed module has been modified by comparing
-/// file hash in the module with the file hashes provided
-/// as an argument to see if module hash is included in the
-/// hashes provided. We only consider file hashes from modified
-/// files.
+/// Checks if a parsed module is modified by getting
+/// the module's file path and checking if it's included
+/// in the set of modified file paths.
 fn is_typed_mod_modified(
     mdef: &ModuleDefinition,
     modified_files: &BTreeSet<PathBuf>,
-    file_hashes: Arc<BTreeMap<PathBuf, FileHash>>,
+    file_paths: Arc<BTreeMap<FileHash, PathBuf>>,
 ) -> bool {
-    !hash_included_in_file_hashes(mdef.loc.file_hash(), modified_files, file_hashes)
+    // unwrap is safe as file paths comes from mapped files
+    // that the compiler relies on when comppiling all modules
+    let mod_file_path = file_paths.get(&mdef.loc.file_hash()).unwrap();
+    modified_files.contains(mod_file_path)
```

Comment:
Nit, wondering if it would be nicer/less possible panick prone to do something like this

```suggestion
    // unwrap is safe as file paths comes from mapped files
    // that the compiler relies on when comppiling all modules
    file_paths.get(&mdef.loc.file_hash()).is_some_and(|mfp| modified_files.contains(mfp))
```